src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/MergeDataToSearchResultOperation.java

```diff
@@ -80,11 +80,12 @@ private SearchResultViewColumn createSearchResultViewColumn(final SearchResultFi
     }
 
     private boolean filterDistinctFieldsByRole(final HashSet<String> addedFields, final SearchResultField resultField) {
-        if (addedFields.contains(resultField.getCaseFieldId())) {
+        String id = buildCaseFieldId(resultField);
```

Comment:
rather than `buildCaseFieldId` consider moving this to a method in the `SearchResultField` class itself for future reuse in other places as well. For example getPath or getFullId.